site/src/templates/exercise.js

```diff
@@ -1,66 +1,130 @@
-import React from 'react'
-import { Link, graphql } from 'gatsby'
+import React from 'react';
+import { Link, graphql } from 'gatsby';
 
-import Layout from '../components/layout'
-import Seo from '../components/seo'
+import Layout from '../components/layout';
+import Seo from '../components/seo';
 
-const BlogPostTemplate = ({ data, pageContext, location }) => {
-  const post = data.markdownRemark
-  const siteTitle = data.site.siteMetadata.title
-  const { previous, next } = pageContext
+function nextPrev(nodes, location) {
+  const len = nodes.length;
+  const current = nodes.findIndex((node) =>
+    location.pathname.includes(node.fields.slug)
+  );
+  const prevNode = current !== 0 ? nodes[current - 1] : false;
+  const nextNode = current !== len - 1 ? nodes[current + 1] : false;
+  return [nextNode, prevNode];
+}
+
+const ExerciseTemplate = ({ children, data, pageContext, location }) => {
+  const pageTitle = pageContext.frontmatter.title;
+  const toc = data.mdx.tableOfContents.items;
+  const nodes = data.allMdx.nodes;
+  const [next, previous] = nextPrev(nodes, location);
 
   return (
-    <Layout location={location} title={siteTitle}>
-      <Seo
-        title={post.frontmatter.title}
-        description={post.frontmatter.description || post.excerpt}
-      />
+    <Layout location={location}>
+      <Seo title={pageTitle} description={pageContext.description} />
       <article className="exercise-main content">
-        <header>
-          <h2>{post.frontmatter.title}</h2>
-        </header>
-        <section dangerouslySetInnerHTML={{ __html: post.html }} />
+        <nav className="nav exercise-nav">
```

Comment:
Very much appreciate this - only suggestion would to see if we can split numbering on circuitpython L1 and L2, it currently goes 1-6 then 1-4.

<img width="297" alt="Screenshot 2023-12-27 at 8 23 15 AM" src="https://github.com/morganstanley/cpx-training/assets/147945654/af4a163c-f522-49a5-96a5-6a7b22919691">

Reply:
Added level numbers:

![Screenshot 2023-12-27 at 12 44 10](https://github.com/morganstanley/cpx-training/assets/414934/f3836641-e689-4f44-b6fb-b9ad0d316106)